app/models/taxon_lineage.rb

```diff
@@ -159,6 +159,21 @@ def self.fill_lineage_details(tax_map, pipeline_run_id)
     tax_map
   end
 
+  def to_a
+    [species_taxid, genus_taxid, family_taxid, order_taxid, class_taxid, phylum_taxid,
```

Comment:
Could the following help with maintainability?
```
def to_a
  self.names_a.map { |col| self[col] }
end
```
and
```
def self.null_array
  TaxonLineage.column_defaults.values_at(*self.names_a)
end
```
That way, the list of columns is only hardcoded in names_a. I think it would be preferable

Reply:
good point.